translate/compiler.go

```diff
@@ -16,30 +16,41 @@
 
 package translate
 
-import "github.com/DE-labtory/koa/ast"
+import (
+	"github.com/DE-labtory/koa/abi"
+	"github.com/DE-labtory/koa/ast"
+	"github.com/DE-labtory/koa/encoding"
+	"github.com/DE-labtory/koa/opcode"
+)
 
-// TODO: parameter should be ast.Contract
 // TODO: implement me w/ test cases :-)
 // CompileContract() compiles a smart contract.
 // returns bytecode and error.
 func CompileContract(c ast.Contract) (Bytecode, error) {
 	bytecode := &Bytecode{
 		RawByte: make([]byte, 0),
 		AsmCode: make([]string, 0),
+		Abi: abi.ABI{
+			Methods: make([]abi.Method, 0),
+		},
 	}
 
 	for _, f := range c.Functions {
 		if err := compileFunction(*f, bytecode); err != nil {
 			return *bytecode, err
```

Comment:
Why return bytecode else?

Reply:
@boohyunsik 
Your question means if error occurs, why return `*bytecode, err`, not `Bytecode{}, err`??
To be easier to debug or test, it returns the compiled bytecode so far.